pkg/internal/checkup/checkup.go

```diff
@@ -975,11 +972,83 @@ func (c *Checkup) checkVMIHotplugVolume(ctx context.Context, errStr *string) err
 	return nil
 }
 
-type checkVMIStatusFn func(*kvcorev1.VirtualMachineInstance) (done bool, err error)
+func (c *Checkup) checkConcurrentVMIBoot(ctx context.Context, errStr *string) error {
+	numOfVMs := c.checkupConfig.NumOfVMs
+	log.Printf("checkConcurrentVMIBoot numOfVMs:%d", numOfVMs)
 
-func (c *Checkup) waitForVMIStatus(ctx context.Context, checkMsg string, result, errStr *string, checkVMIStatus checkVMIStatusFn) error {
-	vmName := c.vmUnderTest.Name
+	if c.goldenImagePvc == nil && c.goldenImageSnap == nil {
+		log.Print(MessageSkipNoGoldenImage)
+		c.results.ConcurrentVMBoot = MessageSkipNoGoldenImage
+		return nil
+	}
+
+	var wg sync.WaitGroup
+	isBootOk := true
+
+	for i := 0; i < numOfVMs; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+
+			vmName := uniqueVMName()
+			log.Printf("Creating VM %q", vmName)
+			vm := newVMUnderTest(vmName, c.goldenImagePvc, c.goldenImageSnap, c.checkupConfig, true)
+			if _, err := c.client.CreateVirtualMachine(ctx, c.namespace, vm); err != nil {
+				log.Printf("failed to create VM %q: %s", vmName, err)
+				isBootOk = false
+				return
+			}
+
+			defer func() {
+				if err := c.client.DeleteVirtualMachine(ctx, c.namespace, vmName); err != nil {
+					log.Printf("failed to delete VM %q: %s", vmName, err)
+				}
+			}()
+
+			var result, errs string
+			if err := c.waitForVMIBoot(ctx, vmName, &result, &errs); err != nil || errs != "" {
+				log.Printf("failed waiting for VM boot %q", vmName)
+				isBootOk = false
+			}
+		}()
+	}
+
+	wg.Wait()
+	if !isBootOk {
+		log.Print(ErrBootFailedOnSomeVMs)
+		c.results.ConcurrentVMBoot = ErrBootFailedOnSomeVMs
```

Comment:
Maybe also keep track of the failed VMs and report how many of them errored

Reply:
It's not useful imho, as we are not interested in partial success, and you have already have each VMI boot status in the log anyway.